File: momhelp-frontend/momhelp-frontend/src/services/vegetableService.js
import api from './api';

const vegetableService = {
  // Get all vegetables
  getAllVegetables: () => {
    return api.get('/vegetables/all');
  },

  // Get vegetable by ID
  getVegetableById: (id) => {
    return api.get(`/vegetables/${id}`);
  },

  // Add new vegetable
  addVegetable: (vegetableData) => {
    return api.post('/vegetables/add', vegetableData);
  },

  // Update vegetable
  updateVegetable: (id, vegetableData) => {
    return api.put(`/vegetables/update/${id}`, vegetableData);
  },

  // Delete vegetable
  deleteVegetable: (id) => {
    return api.delete(`/vegetables/delete/${id}`);
  },

  // Search vegetables by name
  searchVegetables: (name) => {
    return api.get(`/vegetables/search?name=${encodeURIComponent(name)}`);
  },

  // Get available vegetables (not expired)
  getAvailableVegetables: () => {
    return api.get('/vegetables/available');
  },

  // Get spoiled vegetables
  getSpoiledVegetables: () => {
    return api.get('/vegetables/spoiled');
  },

  // Mark vegetable as spoiled
  markAsSpoiled: (id) => {
    return api.put(`/vegetables/mark-spoiled/${id}`);
  },

  // Remove spoiled vegetable
  removeSpoiledVegetable: (id) => {
    return api.delete(`/vegetables/remove-spoiled/${id}`);
  },
};

export default vegetableService;

File: momhelp/src/main/java/com/momhelp/service/VegetableService.java
package com.momhelp.service;

import com.momhelp.dto.VegetableRequestDTO;
import com.momhelp.dto.VegetableResponseDTO;
import java.util.List;

public interface VegetableService {
    
    // Add new vegetable
    VegetableResponseDTO addVegetable(VegetableRequestDTO vegetableRequestDTO);
    
    // Update existing vegetable
    VegetableResponseDTO updateVegetable(Long id, VegetableRequestDTO vegetableRequestDTO);
    
    // Delete vegetable by ID
    void deleteVegetable(Long id);
    
    // Get vegetable by ID
    VegetableResponseDTO getVegetableById(Long id);
    
    // Get all vegetables
    List<VegetableResponseDTO> getAllVegetables();
    
    // Search vegetables by name
    List<VegetableResponseDTO> searchVegetablesByName(String name);
    
    // Get available vegetables (not expired)
    List<VegetableResponseDTO> getAvailableVegetables();
    
    // Get spoiled vegetables
    List<VegetableResponseDTO> getSpoiledVegetables();
    
    // Mark vegetable as spoiled
    VegetableResponseDTO markAsSpoiled(Long id);
    
    // Remove spoiled vegetable
    void removeSpoiledVegetable(Long id);
    
 // Add this method signature
    VegetableResponseDTO useVegetable(Long id, Double weightUsed);
}